src/coverlet.core/Helpers/InstrumentationHelper.cs

```diff
@@ -90,6 +93,34 @@ public static bool HasPdb(string module)
             }
         }
 
+        public static bool EmbeddedPortablePdbHasLocalSource(string module)
+        {
+            using (FileStream moduleStream = File.OpenRead(module))
+            using (var peReader = new PEReader(moduleStream))
+            {
+                foreach (DebugDirectoryEntry entry in peReader.ReadDebugDirectory())
+                {
+                    if (entry.Type == DebugDirectoryEntryType.EmbeddedPortablePdb)
+                    {
+                        using (MetadataReaderProvider embeddedMetadataProvider = peReader.ReadEmbeddedPortablePdbDebugDirectoryData(entry))
+                        {
+                            MetadataReader metadataReader = embeddedMetadataProvider.GetMetadataReader();
+                            foreach (DocumentHandle docHandle in metadataReader.Documents)
+                            {
+                                Document document = metadataReader.GetDocument(docHandle);
+                                string docName = metadataReader.GetString(document.Name);
+
+                                // We stop at first document we need only to verify if module has got local source
+                                return File.Exists(docName);
```

Comment:
@onovotny does it make sense?
You've more familiarity than me with ppdb, I mean is it possible some type of document different than local source code that could return false negative?

Reply:
I'm not sure, sorry.

Reply:
Thanks for quick response, I'll refine with only cs sources.